src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -2312,3 +2312,104 @@ plugin could be rewritten like this:
+always define color variables in the `_bootstrap-defaults.scss` and
+use the variable instead of hard coded color values so that changes to color
+are reflected in similar UI elements.
+
+The main color for the Ceph Dashboard is `$primary`. The primary color is
+used in navigation components and as the `border-color` for input components of
+form.
+
+The secondary color is `$secondary` and is the background color for Ceph
+Dashboard.
+
+Buttons
+~~~~~~~
+Buttons are used for performing actions such as: “Submit”, “Edit, “Create" and
+“Update”. These actions almost always occur on the same page.
+
+**Forms:** When using to submit forms anywhere in the dashboard, the main action
+button should of contrasting color and the secondary button should be of white
+color. The text on the action button should be same as the form title and follow
+a title case. The text on the secondary button should be `Cancel`. Perform the
+action button should always be on right while `Cancel` button should always be
+on left.
+
+**Modals**: The main button should be of the same color as the main action
+button and the secondary action button should be white. The text on the action
+button should follow a title case and correspond to the action to be performed.
+The text on the secondary button should be `Close`.
+
+**Disclosure Button:** Disclosure buttons should be used to allow users to
+display and hide additional content in the interface.
+
+**Action Button**: Use action button to perform actions such as edit or update a
+component. The button color should be same as the form's main button color. The
+text on button should follow title case and have an icon corresponding to the
+actions they perform.
+
+**Drop Down Buttons:** Use dropdown buttons to display predefined lists of
+actions. All drop down buttons have icons corresponding to the action they
+perform.  
+
+Links
+~~~~~
+Use text hyperlinks as navigation to guide users to a new page in the application
+or to anchor users to a section within a page. The color of the hyperlinks
+should be `$primary`.
+
+Tabs
+~~~~
+The current opened tab should have black font color and the link to other tabs
+should have `$primary` font.
+
+Forms
+~~~~~
+Mark invalid form fields with red outline and show a meaningful error message.
+Use red as font color for message and be as specific as possible.
+Mark valid forms with a green outline and a green tick at the end of the form.
+
+Modals
+~~~~~~
+Blur any interface elements in the background to bring the modal content into
+focus. The heading of the modal should reflect the action it can perform and
+should be clearly mentioned at the top of the modal.
+The button to close the modal should be in the footer.
+
+Icons
+~~~~~
+We use Font Awesome classes for icons.
+When icons are next to text, they should be center-aligned horizontally. If icons
+are stacked, they should also be center-aligned vertically.
+Use small icons with buttons. For notificatons use large icons.
```

Comment:
We should mention that we have a list of used icons in `src/app/shared/enum/icons.enum.ts`, these should be referenced in the HTML, so its easier to change them later.